bin/restore-standalone.sh

```diff
@@ -0,0 +1,235 @@
+  rm -r "$WORKING_DIR"
+  echo ""
+}
+
+trap _term SIGTERM
+trap _term INT
+
+##################################### Parse arguments
+
+OPTIONS="h"
+LONGOPTS=bootstrap-server:,source-dir:,topics:,batch-size:,offset-file:,command-config:,help,debug
+
+HELP=$(
+  cat <<END
+--bootstrap-server  [REQUIRED] The Kafka server to connect to
+--source-dir        [REQUIRED] Directory where the backup files are found
+--topics            [REQUIRED] List of topics to restore
+--batch-size                   Batch size (Default: 1MiB)
+--offset-file                  File where to store offsets. THIS FILE IS CRUCIAL FOR A CORRECT RESTORATION PROCESS
+                               IF YOU LOSE IT YOU NEED TO START THE BACKUP FROM SCRATCH. OTHERWISE YOU WILL HAVE DUPLICATE DATA
+                               Default: [source-dir]/restore.offsets
+--command-config    <FILE>     Property file containing configs to be
+                               passed to Admin Client. Only useful if you have additional connection options
+--help                         Prints this message
+--debug                        Print Debug information
+END
+)
+
+BOOTSTRAP_SERVER=""
+SOURCE_DIR=""
+TOPICS=""
+OFFSET_FILE=""
+BATCH_SIZE="$((1 * 1024 * 1024))"
+COMMAND_CONFIG=""
+PLUGIN_PATH="$(dirname "${BASH_SOURCE[0]}")"
+CONNECT_BIN=""
+DEBUG="n"
+
+# -temporarily store output to be able to check for errors
+# -activate quoting/enhanced mode (e.g. by writing out “--options”)
+# -pass arguments only via   -- "$@"   to separate them correctly
+! PARSED=$(getopt --options=$OPTIONS --longoptions=$LONGOPTS --name "$0" -- "$@")
+if [[ ${PIPESTATUS[0]} -ne 0 ]]; then
+  # e.g. return value is 1
+  #  then getopt has complained about wrong arguments to stdout
+  exit 2
+fi
+# read getopt’s output this way to handle the quoting right:
+eval set -- "$PARSED"
+
+while true; do
+  case "$1" in
+  --bootstrap-server)
+    BOOTSTRAP_SERVER="$2"
+    shift 2
+    ;;
+  --source-dir)
+    SOURCE_DIR="$2"
+    shift 2
+    ;;
+  --topics)
+    TOPICS="$2"
+    shift 2
+    ;;
+  --batch-size)
+    BATCH_SIZE="$2"
+    shift 2
+    ;;
+  --offset-file)
+    OFFSET_FILE="$2"
+    shift 2
+    ;;
+  --command-config)
+    COMMAND_CONFIG="$2"
+    shift 2
+    ;;
+  -h | --help)
+    echo "$HELP"
+    exit 0
+    ;;
+  -d | --debug)
+    DEBUG=y
+    shift
+    ;;
+  --)
+    shift
+    break
+    ;;
+  *)
+    echo "$1 $2 Programming error"
+    exit 3
+    ;;
+  esac
+done
+
+##################################### Check arguments
+
+if [ -z "$BOOTSTRAP_SERVER" ]; then
+  echo "--bootstrap-server is missing"
+  echo "$HELP"
+  exit 1
+fi
+
+if [ -z "$SOURCE_DIR" ]; then
+  echo "--source-dir is missing"
+  echo "$HELP"
+  exit 1
+fi
+
+if [ ! -d "$SOURCE_DIR" ]; then
+  echo "Directory $SOURCE_DIR does not exist."
+  exit 1
+fi
+
+if [ -z "$TOPICS" ]; then
+  echo "--topics is missing"
+  echo "$HELP"
+  exit 1
+fi
+
+if [ -z "$OFFSET_FILE" ]; then
+  OFFSET_FILE="$SOURCE_DIR/restore.offsets"
```

Comment:
I see no usage of this file later in configs.. Is it expected?

Reply:
Good catch. Thank you!